### PR TITLE
Support both ember-concurrency 1.x and 2.x

### PR DESCRIPTION
**Pull request by maxfierke**
We've just released ember-concurrency 2.0.0-beta.1. It's a largely API-compatible re-engineering of ember-concurrency's internals to support tracked properties and decouple from Ember (shedding EmberObject, etc.). As a popular library which depends on ember-concurrency, it would be nice for `ember-power-select` to support both 1.x and 2.x versions concurrently so that folks can upgrade the version of ember-concurrency in their applications at their leisure without waiting for all the add-ons they use to switch over to a newer version (and risk leaving folks who cannot upgrade yet behind.)

This PR adds some scenarios to ember-try to enable testing against both major releases and relaxes the version restriction.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (simonihmig) [inline .github/workflows/ci.yml:68]: Do we really need two separate scenarios here? Like all the other scenarios are already using one of the versions, so we really just need one additional scenario to test with the other version, don't we?

Our test matrix is already quite big, and in this case two additional scenarios cause 4 additional test jobs, as each is running both for Bootstrap 3 and 4.

File: config/ember-try.js
Comment (simonihmig) [inline config/ember-try.js:122]:
```suggestion
            'ember-concurrency': '^2.0.0-beta.1',
            bootstrap: bootstrapVersion,
```

File: config/ember-try.js
Comment (simonihmig) [inline config/ember-try.js:114]: The two scenarios are failing for Bootstrap 3, probably due to this one missing (see other scenarios):

```suggestion
            'ember-concurrency': '^1.3.0',
            bootstrap: bootstrapVersion,
```

**Review by simonihmig — COMMENTED** [review]
Thanks for pro-actively working on this, @maxfierke! 

This is a good change, I remember the churn it caused when e-c was updating its 0.x minor, and causing conflicts everywhere...

Some tests are failing, but I think easy to fix, see me comments.

File: package.json
Comment (maxfierke) [inline package.json:57]: messed up the version constraint here... should be `>=1.3.0 <3`. will update

File: .github/workflows/ci.yml
Comment (maxfierke) [reply .github/workflows/ci.yml:68]: Not really, but it does protect against losing coverage if the version in `yarn.lock` changes. I can limit it to just the `ember-concurrency-2.x` add, though.

Comment (jelhan) [reply .github/workflows/ci.yml:68]: I think we should enforce a default version explicitly using yarn resolutions and only cover the other one with an explicit scenario.

I would be fine with using v2 as the default one starting from now. Than we can drop support for v1 in the next major.

Comment (simonihmig) [reply .github/workflows/ci.yml:68]: With the current version specifier in package.json, e-c 2.0.0 *would* get used by default if released, but not 2.0.0-beta.x, right? Or at least we need to update `yarn.lock`.

So can we just wait for the stable release, and then drop the `ember-concurrency-2.x` scenario?

Comment (maxfierke) [reply .github/workflows/ci.yml:68]: I'll keep the resolved version pinned at 1.x and remove the specific test setting for 1.x, and leave it up to your discretion on when to switch to 2.x

**Review by simonihmig — APPROVED** [review]
Thanks again! :+1:

**Pull request merged**
